lib/active_model/serializer/collection_serializer.rb

```diff
@@ -10,8 +10,14 @@ class CollectionSerializer
       def initialize(resources, options = {})
         @root = options[:root]
         @object = resources
+
+        serializer_context_class = options.fetch(:serializer_context_class, ActiveModel::Serializer)
+
+        if resources.blank? && options[:serializer]
+          @each_serializer = options[:serializer]
```

Comment:
`@each_serializer` is too much linked to the controller option name IMO. I'd rather call it `@resource_serializer` or simply `@serializer`.